src/program/lwaftr/tests/subcommands/monitor_test.py

```diff
@@ -17,37 +16,26 @@
 
 @unittest.skipUnless(SNABB_PCI0, 'NIC not configured')
 @unittest.skipUnless(TAP_IFACE, tap_err_msg)
-class TestMonitor(unittest.TestCase):
+class TestMonitor(BaseTestCase):
 
-    run_cmd_args = (
-        SNABB_CMD, 'lwaftr', 'run',
-        '--name', 'monitor_test',
+    daemon_args = (
+        str(SNABB_CMD), 'lwaftr', 'run',
         '--bench-file', '/dev/null',
-        '--conf', DATA_DIR / 'icmp_on_fail.conf',
+        '--conf', str(DATA_DIR / 'icmp_on_fail.conf'),
         '--on-a-stick', SNABB_PCI0,
         '--mirror', TAP_IFACE,
     )
-
-    monitor_cmd_args = (
-        SNABB_CMD, 'lwaftr', 'monitor',
-        '--name', 'monitor_test',
-        'all',
-    )
-
-    # Use setUpClass to only setup the "run" daemon once for all tests.
-    @classmethod
-    def setUpClass(cls):
-        cls.run_cmd = sh.sudo(*cls.run_cmd_args, _bg=True)
+    monitor_args = (str(SNABB_CMD), 'lwaftr', 'monitor', 'all')
+    wait_for_daemon_startup = True
 
     def test_monitor(self):
-        output = sh.sudo(*self.monitor_cmd_args)
-        self.assertEqual(output.exit_code, 0)
-        self.assertIn('Mirror address set', output)
-        self.assertIn('255.255.255.255', output)
-
-    @classmethod
-    def tearDownClass(cls):
-        cls.run_cmd.terminate()
+        monitor_args = list(self.monitor_args)
+        monitor_args.append(str(self.daemon.pid))
+        output = self.run_cmd(monitor_args)
+        self.assertIn(b'Mirror address set', output,
+            b'\n'.join((b'OUTPUT', output)))
```

Comment:
Not sure what you mean: it *is* asserted in this `test_monitor` method, and only here. I removed the `setupClass` and `tearDownClass` methods from this `TestCase`, because it now relies on `BaseTestCase`.